devcycle_python_sdk/managers/event_queue_manager.py

```diff
@@ -114,4 +125,68 @@ def run(self):
 
     def close(self):
         self._processing_enabled = False
+        try:
+            self._flush_events()
+        except Exception as e:
+            logger.info(f"DVC Error flushing events when closing client: {str(e)}")
         self.join(timeout=1)
+
+    def queue_event(self, user: User, event: Event) -> None:
+        if user is None:
+            raise ValueError("user cannot be None")
+
+        if event is None:
+            raise ValueError("event cannot be None")
+
+        if not event.type:
+            raise ValueError("event type not set")
+
+        try:
+            self._check_queue_status()
+        except QueueFullError:
+            logger.warning(f"Event queue is full, dropping event: {event}")
+            return
+
+        user_json = json.dumps(user.to_json())
+        event_json = json.dumps(event.to_json())
+        self._local_bucketing.queue_event(user_json, event_json)
+
+    def queue_aggregate_event(
+        self, event: Event, bucketed_config: Optional[BucketedConfig]
+    ) -> None:
+        if event is None:
+            raise ValueError("event cannot be None")
+
+        if not event.type:
+            raise ValueError("event type not set")
+
+        if not event.target:
+            raise ValueError("event target not set")
+
+        try:
+            self._check_queue_status()
+        except QueueFullError:
+            logger.warning("Event queue is full, dropping event")
+            return
+
+        event_json = json.dumps(event.to_json())
+        if bucketed_config and bucketed_config.variable_variation_map:
+            variation_map_json = json.dumps(bucketed_config.variable_variation_map)
+        else:
+            variation_map_json = "{}"
+        self._local_bucketing.queue_aggregate_event(event_json, variation_map_json)
+
+    def _check_queue_status(self) -> None:
+        if self._flush_needed() and not self._flush_lock.locked():
```

Comment:
I think this is a redundant check of `_flush_lock.locked()`? It'll just check it in `_flush_events` anyways.

Reply:
true.  i'll remove the extra check